From fabacf42c27e225da01cd5d1f225702d214515d2 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Thu, 24 Jun 2021 07:31:31 -0500
Subject: [PATCH 2/2] rzg: drivers: pfc: Support Beacon Boards

There are few GPIO configuration changes needed for the Beacon
boards.  Insert those changes only when the BOARD is configured
for BEACON without breaking backwards compatibility

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 drivers/renesas/rzg/pfc/G2H/pfc_init_g2h.c | 16 ++++++++++++++++
 drivers/renesas/rzg/pfc/G2M/pfc_init_g2m.c | 16 ++++++++++++++++
 drivers/renesas/rzg/pfc/G2N/pfc_init_g2n.c | 16 ++++++++++++++++
 3 files changed, 48 insertions(+)

diff --git a/drivers/renesas/rzg/pfc/G2H/pfc_init_g2h.c b/drivers/renesas/rzg/pfc/G2H/pfc_init_g2h.c
index 90a1c9988..46d8bc0e6 100644
--- a/drivers/renesas/rzg/pfc/G2H/pfc_init_g2h.c
+++ b/drivers/renesas/rzg/pfc/G2H/pfc_init_g2h.c
@@ -951,6 +951,15 @@ void pfc_init_g2h(void)
 		      POC_SD0_DAT2_33V |
 		      POC_SD0_DAT1_33V |
 		      POC_SD0_DAT0_33V |
+#if (BOARD == BEACON)
+		      POC_SD1_DAT3_33V |
+		      POC_SD1_DAT2_33V |
+		      POC_SD1_DAT1_33V |
+		      POC_SD1_DAT0_33V |
+		      POC_SD1_CMD_33V  |
+		      POC_SD1_CLK_33V  |
+		      POC_SD0_DAT3_33V |
+#endif
 		      POC_SD0_CMD_33V |
 		      POC_SD0_CLK_33V);
 
@@ -1302,9 +1311,16 @@ void pfc_init_g2h(void)
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
 	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
 	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
+#if (BOARD == BEACON)
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00000F00U);
+#else
 	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
 	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
 	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
 	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
+#endif
 	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
 }
diff --git a/drivers/renesas/rzg/pfc/G2M/pfc_init_g2m.c b/drivers/renesas/rzg/pfc/G2M/pfc_init_g2m.c
index f76b83f9a..f85c863ea 100644
--- a/drivers/renesas/rzg/pfc/G2M/pfc_init_g2m.c
+++ b/drivers/renesas/rzg/pfc/G2M/pfc_init_g2m.c
@@ -990,6 +990,15 @@ void pfc_init_g2m(void)
 		      | POC_SD0_DAT1_33V
 		      | POC_SD0_DAT0_33V
 		      | POC_SD0_CMD_33V
+#if (BOARD == BEACON)
+		      | POC_SD1_DAT3_33V
+		      | POC_SD1_DAT2_33V
+		      | POC_SD1_DAT1_33V
+		      | POC_SD1_DAT0_33V
+		      | POC_SD1_CMD_33V
+		      | POC_SD1_CLK_33V
+		      | POC_SD0_DAT3_33V
+#endif
 		      | POC_SD0_CLK_33V);
 
 	/* initialize DRV control register */
@@ -1291,10 +1300,17 @@ void pfc_init_g2m(void)
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
 	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
 	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
+#if (BOARD == BEACON)
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00000F00U);
+#else
 	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
 	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
 	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
 	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
+#endif
 	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
 
 }
diff --git a/drivers/renesas/rzg/pfc/G2N/pfc_init_g2n.c b/drivers/renesas/rzg/pfc/G2N/pfc_init_g2n.c
index c951e0a58..9ac004a07 100644
--- a/drivers/renesas/rzg/pfc/G2N/pfc_init_g2n.c
+++ b/drivers/renesas/rzg/pfc/G2N/pfc_init_g2n.c
@@ -947,6 +947,15 @@ void pfc_init_g2n(void)
 		      POC_SD0_DAT2_33V |
 		      POC_SD0_DAT1_33V |
 		      POC_SD0_DAT0_33V |
+#if (BOARD == BEACON)
+		      POC_SD1_DAT3_33V |
+		      POC_SD1_DAT2_33V |
+		      POC_SD1_DAT1_33V |
+		      POC_SD1_DAT0_33V |
+		      POC_SD1_CMD_33V  |
+		      POC_SD1_CLK_33V  |
+		      POC_SD0_DAT3_33V |
+#endif
 		      POC_SD0_CMD_33V |
 		      POC_SD0_CLK_33V);
 
@@ -1298,9 +1307,16 @@ void pfc_init_g2n(void)
 	mmio_write_32(GPIO_INOUTSEL0, 0x00000001U);
 	mmio_write_32(GPIO_INOUTSEL1, 0x00100B00U);
 	mmio_write_32(GPIO_INOUTSEL2, 0x00000418U);
+#if (BOARD == BEACON)
+	mmio_write_32(GPIO_INOUTSEL3, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL4, 0x00000000U);
+	mmio_write_32(GPIO_INOUTSEL5, 0x00000008U);
+	mmio_write_32(GPIO_INOUTSEL6, 0x00000F00U);
+#else
 	mmio_write_32(GPIO_INOUTSEL3, 0x00002000U);
 	mmio_write_32(GPIO_INOUTSEL4, 0x00000040U);
 	mmio_write_32(GPIO_INOUTSEL5, 0x00000208U);
 	mmio_write_32(GPIO_INOUTSEL6, 0x00013F00U);
+#endif
 	mmio_write_32(GPIO_INOUTSEL7, 0x00000003U);
 }
-- 
2.17.1

